standalone/content/js/standaloneRoomViews.jsx

```diff
@@ -522,6 +597,73 @@ loop.standaloneRoomViews = (function(mozL10n) {
     }
   });
 
+  var StandaloneNotificationView = React.createClass({
+    propTypes: {
+      description: React.PropTypes.string.isRequired,
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher),
+      peerDisconnected: React.PropTypes.bool,
+      thumbnail: React.PropTypes.string,
+      url: React.PropTypes.string
+    },
+
+    /**
+     * Dispatches an action to record when the link is clicked.
+     */
+    handleLinkClick: function() {
+      this.props.dispatcher.dispatch(new sharedActions.RecordClick({
+        linkInfo: "Shared URL"
+      }));
+    },
+
+    render: function() {
+      // Bug 1196143 - formatURL sanitizes(decodes) the URL from IDN homographic attacks.
+      // Try catch to not produce output if invalid url
+      try {
+        var sanitizeURL = loop.shared.utils.formatURL(this.props.url, true).hostname;
+      } catch (ex) {
+        return null;
+      }
+
+      var thumbnail = this.props.thumbnail;
+
+      if (!thumbnail) {
+        thumbnail = "shared/img/icons-16x16.svg#globe";
+      }
+
+      var title = this.props.peerDisconnected ?
+        "Your friend has left" : mozL10n.get("rooms_only_occupant_label2");
+
+      return (
+        <div className="room-notification-area">
+          <div className="room-notification-header">
+            {title}
+          </div>
+          <div className="room-notification-content">
+            <p>
+              While you wait, you can review what your friend wants to show you in another tab. We will notify you when they join:
+            </p>
+
+            <div className="room-notification-context">
+              <a className="context-wrapper"
+                 href={this.props.url ? this.props.url : null}
+                 onClick={this.handleLinkClick}
+                 rel="noreferrer"
+                 target="_blank">
+                <img className="context-preview" src={thumbnail} />
+                <span className="context-info">
+                  {this.props.description}
+                  <span className="context-url">
+                    {sanitizeURL}
+                  </span>
+                </span>
+              </a>
+            </div>
+          </div>
+        </div>
+      );
+    }
+  });
+
   var StandaloneRoomView = React.createClass({
     mixins: [
       Backbone.Events,
```

Comment:
There are hardcoded strings and no unit tests, was everything pushed in the commit?